met-api/src/met_api/services/submission_service.py

```diff
@@ -140,6 +152,75 @@ def validate_review(cls, values: dict, user):
            not rejected_reason_other:
             raise ValueError('A rejection reason is required.')
 
+    @classmethod
+    def add_or_update_staff_note(cls, survey_id, submission_id, staff_notes):
+        """Process staff note for a comment."""
+        for staff_note in staff_notes:
+            note = StaffNote.get_staff_note_by_type(submission_id, staff_note.get('note_type'))
+            if note:
+                note.note = staff_note['note']
+                StaffNote.flush(note)
+            else:
+                doc = SubmissionService._create_staff_notes(survey_id, submission_id, staff_note)
+                StaffNote.flush(doc)
+
+    @staticmethod
+    def _create_staff_notes(survey_id, submission_id, staff_note):
+        doc: StaffNote = StaffNote()
+        doc.note = staff_note['note']
+        doc.note_type = staff_note['note_type']
+        doc.survey_id = survey_id
+        doc.submission_id = submission_id
+        return doc
+
+    @staticmethod
+    def _should_send_email(submission_id: int, staff_comment_details: dict) -> bool:
+        """Check if an email should be sent for a rejected submission."""
+        # Dont send the mail
+        #   if the comment has threat
+        #   if notify_email is false
+        # Send the mail
+        #   if the status of the comment is rejected
+        #   if review note has not changed
+
+        if staff_comment_details.get('has_threat') is True:
+            return False
+        if staff_comment_details.get('notify_email') is False:
+            return False
+        if staff_comment_details.get('status_id') == Status.Rejected.value:
+            return True
```

Comment:
i think you need to remove the return on this line, this will return True if status == rejected and it won't check if the note has changed e.g.

if status == rejected AND note_changed
 return true

Reply:
Thanks have changed it to check the review not or reasons within this.